featurebyte/service/entity_validation.py

```diff
@@ -74,17 +75,27 @@ def _get_original_serving_name(serving_name: str) -> str:
                 return inv_serving_names_mapping[serving_name]
             return serving_name
 
-        planner = FeatureExecutionPlanner(
-            graph=graph,
-            is_online_serving=False,
-            serving_names_mapping=serving_names_mapping,
-        )
-        plan = planner.generate_plan(nodes)
+        # Infer provided entities from provided request columns
         candidate_serving_names = {_get_original_serving_name(col) for col in request_column_names}
         provided_entities = await self.entity_service.get_entities_with_serving_names(
             candidate_serving_names
         )
-        required_entities = await self.entity_service.get_entities(plan.required_entity_ids)
+
+        # Extract required entities from feature list (faster) or graph
+        if feature_list_model is not None:
+            required_entities = await self.entity_service.get_entities(
+                set(feature_list_model.entity_ids)
+            )
```

Comment:
Just to check we want `entity_ids` here but not `primary_entity_ids`. `entity_ids` is the union of individual feature's `entity_ids`.

Reply:
Yes, that is intended for now. That returns the same set of entities as the "else" branch below. This will transition to using `primary_entity_ids` once sql generation supports using feature level frozen relationships.